src/views/DocumentationTopic.vue

```diff
@@ -9,149 +9,66 @@
-        class="full-width-container topic-wrapper"
+        :is="rootLink ? 'router-link' : 'span'"
+        :to="rootLink"
+        class="nav-title-link"
       >
-        <PortalTarget name="modal-destination" multiple />
-        <template #aside="{ scrollLockID, breakpoint }">
-          <NavigatorDataProvider
-            :interface-language="topicProps.interfaceLanguage"
-            :technologyUrl="technology.url"
-            :api-changes-version="store.state.selectedAPIChangesVersion"
-            ref="NavigatorDataProvider"
-          >
-            <template #default="slotProps">
-              <div class="doc-topic-aside">
-                <QuickNavigationModal
-                  v-if="enableQuickNavigation"
-                  :children="slotProps.flatChildren"
-                  :showQuickNavigationModal.sync="showQuickNavigationModal"
-                  :technology="technology.title"
-                />
-                <transition name="delay-hiding">
-                  <Navigator
-                    v-show="sidenavVisibleOnMobile || breakpoint === BreakpointName.large"
-                    :flatChildren="slotProps.flatChildren"
-                    :parent-topic-identifiers="parentTopicIdentifiers"
-                    :technology="slotProps.technology || technology"
-                    :is-fetching="slotProps.isFetching"
-                    :error-fetching="slotProps.errorFetching"
-                    :api-changes="slotProps.apiChanges"
-                    :references="topicProps.references"
-                    :navigator-references="slotProps.references"
-                    :scrollLockID="scrollLockID"
-                    :render-filter-on-top="breakpoint !== BreakpointName.large"
-                    @close="handleToggleSidenav(breakpoint)"
-                  >
-                    <template v-if="enableQuickNavigation" #filter>
-                      <QuickNavigationButton @click.native="openQuickNavigationModal" />
-                    </template>
-                    <template #navigator-head>
-                      <component
-                        :is="rootLink ? 'router-link' : 'span'"
-                        :to="rootLink"
-                      >
-                        {{ $t('documentation.title') }}
-                      </component>
-                    </template>
-                  </Navigator>
-                </transition>
-              </div>
-            </template>
-          </NavigatorDataProvider>
-        </template>
-        <Topic
-          v-bind="topicProps"
-          :key="topicKey"
-          :objcPath="objcPath"
-          :swiftPath="swiftPath"
-          :isSymbolDeprecated="isSymbolDeprecated"
-          :isSymbolBeta="isSymbolBeta"
-          :languagePaths="languagePaths"
-          :enableOnThisPageNav="enableOnThisPageNav"
-          :enableMinimized="enableMinimized"
-          :hierarchyItems="hierarchyItems"
-        />
+        {{ $t('documentation.title') }}
```

Comment:
> I think it's quite confusing for this to remain here. It's unclear what it is for at this level. It's ultimately being pass down quite a few components deeper.

I made it in this way so we could be as flexible enough providing different titles to the views that will use DocumentationLayout.

> We can pretty easily remove all the logic that computes rootLink to the new BaseNavView component, since they are not really used by the Topic component anyways.

`rootLink` depends on `rootHierarchyReference` that depends on `references` coming from `topicProps`, so if I bring `rootLink` to the DocumentationNav, I should bring `topicProps` too, which I'm trying to avoid in this refactor

> move it above the Topic component like it was before

Done!

> give it a better name than title

Do you have any suggestions for the name?

Reply:
I renamed the slot as `nav-title` :)